omc3/plotting/utils/windows.py

```diff
@@ -0,0 +1,208 @@
+"""
+Plot Window
+------------
+
+In this module different classes are defined, allowing to put plots 
+manually into windows.
+These windows are QT-based, created with ``qtpy'' which allows to use 
+either PySide(2 or 6) or PyQt(5 or 6), depending on which it installed on the system.
+As the ``qtpy`` library is optional, there are some checks to make sure 
+the imports do not fail, but then the classes cannot be used.
+To check if QtPy is installed, either run :meth:`omc3.plotting.plot_window.is_qtpy_installed`
+or try to initialize one of the windows, which will fail with a TypeError, 
+as they want to call QApplication which is set to `None`.
+
+An exception to all of this is  :meth:`omc3.plotting.plot_window.create_pyplot_window_from_fig`,
+which allows to create a `pyplot` handled window from an already existing figure.
+This way, figures can be created without manager (which makes them resource friendlier)
+and can either be added to a QT-Window or, if not installed, opened by `pyplot`.
+"""
+import sys
+from typing import Dict, List, Tuple
+
+import matplotlib
+from matplotlib import pyplot as plt, rcParams
+from matplotlib.figure import Figure
+
+from omc3.utils import logging_tools
+
+LOG = logging_tools.get_logger(__name__)
+
+# --- optional qtpy import block -----------------------------------------------
+
+try:
+    from qtpy.QtWidgets import (
+        QApplication, QMainWindow, QTabWidget, QVBoxLayout, QWidget
+    )
+except ImportError as e:
+    LOG.debug(f"Could not import QtPy: {str(e)}")
+    QMainWindow, QApplication, QVBoxLayout  = None, None, None
+    QWidget, QTabWidget = object, object
+else:
+    from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg as FigureCanvas
+    from matplotlib.backends.backend_qt5agg import NavigationToolbar2QT as NavigationToolbar
```

Comment:
Again, Qt5 has reached EOL. The [matplotlib docs](https://matplotlib.org/stable/api/backend_qt_api.html) say that these modules are deprecated and here for backwards-comatibility. I think the recommended way is to use `matplotlib.backends.backend_qtagg` instead and let matplotlib figure out the Qt version and bindings itself.